src/DegenGambit.sol

```diff
@@ -660,37 +697,67 @@ contract DegenGambit is ERC20, ReentrancyGuard {
-    /// @dev msg.sender is assumed to be the player. This call cannot be delegated to a different account.
-    /// @param boost Whether or not the player is using a boost.
-    function spin(bool boost) external payable {
-        uint256 requiredFee = spinCost(msg.sender);
-        if (msg.value < requiredFee) {
+    /// @dev  This call can be delegated to a different account.
+    /// @param boost Whether or not the player is using a boost, msg.sender is paying the boost
+    /// @param spinPlayer account spin is for
+    /// @param streakPlayer account streak reward is for
+    /// @param value value being sent to contract
+    function _spin(
+        address spinPlayer,
+        address streakPlayer,
+        bool boost,
+        uint256 value
+    ) internal {
+        uint256 requiredFee = spinCost(spinPlayer);
+        if (value < requiredFee) {
             revert InsufficientValue();
         }
 
         uint256 currentDay = block.timestamp / SECONDS_PER_DAY;
-        if (LastStreakDay[msg.sender] + 1 == currentDay) {
-            _mint(msg.sender, DailyStreakReward);
-            emit DailyStreak(msg.sender, currentDay);
+        if (LastStreakDay[streakPlayer] + 1 == currentDay) {
+            _mint(streakPlayer, DailyStreakReward);
+            emit DailyStreak(streakPlayer, currentDay);
         }
-        LastStreakDay[msg.sender] = currentDay;
+        LastStreakDay[streakPlayer] = currentDay;
 
         uint256 currentWeek = currentDay / 7;
-        if (LastStreakWeek[msg.sender] + 1 == currentWeek) {
-            _mint(msg.sender, WeeklyStreakReward);
-            emit WeeklyStreak(msg.sender, currentWeek);
+        if (LastStreakWeek[streakPlayer] + 1 == currentWeek) {
+            _mint(streakPlayer, WeeklyStreakReward);
+            emit WeeklyStreak(streakPlayer, currentWeek);
         }
-        LastStreakWeek[msg.sender] = currentWeek;
+        LastStreakWeek[streakPlayer] = currentWeek;
 
         if (boost) {
             // Burn an ERC20 token off of this contract from the player's account.
             _burn(msg.sender, 1);
         }
 
-        LastSpinBlock[msg.sender] = _blockNumber();
-        LastSpinBoosted[msg.sender] = boost;
+        LastSpinBlock[spinPlayer] = _blockNumber();
+        LastSpinBoosted[spinPlayer] = boost;
+
+        emit Spin(spinPlayer, boost);
+    }
+
+    /// Spin the slot machine.
+    /// @notice If the player sends more value than they absolutely need to, the contract simply accepts it into the pot.
+    /// @dev  Assumes msg.sender is player. This call cannot be delegated to a different account.
+    /// @param boost Whether or not the player is using a boost, msg.sender is paying the boost
+    function spin(bool boost) external payable {
+        _spin(msg.sender, msg.sender, boost, msg.value);
+    }
 
-        emit Spin(msg.sender, boost);
+    /// Spin the slot machine.
```

Comment:
Can you update this line on the comment?

Something like "Spin the slot machine for a given account."

With a clarifying comment like... "This method can be used by syndicated clients to play the game on behalf of their players."